trainer/lib/trainer/xcresult/test_plan.rb

```diff
@@ -0,0 +1,89 @@
+require_relative 'test_suite'
+
+module Trainer
+  module XCResult
+    # Represents a collection of test suites + the configuration, and device used to run them
+    class TestPlan
+      attr_reader :test_suites, :name, :configurations, :devices
+      attr_accessor :output_remove_retry_attempts
+
+      def initialize(test_suites:, configurations: [], devices: [], output_remove_retry_attempts: false)
+        @test_suites = test_suites
+        @configurations = configurations
+        @devices = devices
+        @output_remove_retry_attempts = output_remove_retry_attempts
+      end
+
+      def self.from_json(json:)
+        # Extract configurations and devices
+        configurations = json['testPlanConfigurations'] || []
+        devices = json['devices'] || []
+
+        # Find the test plan node (root of test results)
+        test_plan_node = json['testNodes']&.find { |node| node['nodeType'] == 'Test Plan' }
+        return new(test_suites: []) if test_plan_node.nil?
+
+        # Convert test plan node's children (test bundles) to TestSuite objects
+        test_suites = test_plan_node['children']&.map do |test_bundle|
+          TestSuite.from_json(
+            node: test_bundle
+          )
+        end || []
+
+        new(
+          test_suites: test_suites,
+          configurations: configurations,
+          devices: devices
+        )
+      end
+
+      # Allows iteration over test suites. Used by TestParser to collect test results
+      include Enumerable
+      def each(&block)
+        test_suites.map(&:to_hash).each(&block)
+      end
+
+      # Generates a JUnit-compatible XML representation of the test plan
+      # See https://github.com/testmoapp/junitxml/
+      def to_xml
+        # Create the root testsuites element with calculated summary attributes
+        testsuites = Helper.create_xml_element('testsuites',
+          tests: test_suites.sum(&:test_cases_count).to_s,
+          failures: test_suites.sum(&:failures_count).to_s,
+          skipped: test_suites.sum(&:skipped_count).to_s,
+          time: test_suites.sum(&:duration).to_s)
+
+        # Add each test suite to the root
+        test_suites.each do |suite|
+          testsuites.add_element(suite.to_xml(output_remove_retry_attempts: output_remove_retry_attempts))
+        end
+
+        # Convert to XML string with prologue
+        doc = REXML::Document.new
```

Comment:
```suggestion
        doc = REXML::Document.new
        doc.context[:attribute_quote] = :quote
```

See https://github.com/fastlane/fastlane/pull/29463#issuecomment-2704511016

Not sure how much single vs double quote matters for attribute values but I believe this would make it double-quotes to be consistent with what [the legacy junit generator](https://github.com/fastlane/fastlane/blob/alisoftware/trainer/support-swift-testing/trainer/lib/trainer/junit_generator.rb) would produce.

Reply:
The usage of `:attribute_quote` doesn't appear to be explicitly documented :(

But you can see it in action here: https://github.com/ruby/rexml/blob/master/test/test_attributes.rb#L181

Reply:
Nice one!

But since those single vs double quotes were not the core of the issue CircleCI was having with the XML files (as it was instead the `<properties>` not being expected at the top-level element, as you later discovered), and with the fact that `:attribute_quote` is not officially documented, I'd prefer to avoid using it / relying on it in our implementation after all, and stick to the official approach and defaults of how REXML generates and quotes the attributes. wdyt?

Reply:
Agreed, probably not worth the trouble for something that wasn't actually causing problems.